### PR TITLE
service: hid: Improve npad device detection

### PR DESCRIPTION
**Pull request by german77**
This PR should fix random input in some games like Pokemon Sword. Random disconnects on games like Flip Wars.

- Stubs SetPalmaBoostMode
- Signal the update event on AcquireNpadStyleSetUpdateEventHandle
- Initialize and reset shared memory properly instead of fully deleting the contents

### CODE REVIEW

File: app/src/main/cpp/skyline/services/hid/IHidServer.h [hunk 1]
```diff
@@ -178,6 +178,12 @@ namespace skyline::service::hid {
          */
         Result SendVibrationValues(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
 
+        /**
+         * @brief Sets boos mode to a Palma device
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/hid/IHidServer.h:182]: boost*

File: app/src/main/cpp/skyline/input/npad_device.cpp [hunk 2]
```diff
@@ -188,8 +200,8 @@ namespace skyline::input {
 
         info.header.timestamp = util::GetTimeTicks();
         info.header.entryCount = std::min(static_cast<u8>(info.header.entryCount + 1), constant::HidEntryCount);
-        info.header.maxEntry = info.header.entryCount;
-        info.header.currentEntry = (info.header.currentEntry != constant::HidEntryCount - 1) ? info.header.currentEntry + 1 : 0;
+        info.header.maxEntry = info.header.entryCount - 1;
```
Comment (bylaws) [inline app/src/main/cpp/skyline/input/npad_device.cpp:203]: is this needed for touch too?

File: app/src/main/cpp/skyline/input/npad_device.cpp [hunk 3]
```diff
@@ -10,7 +10,19 @@ namespace skyline::input {
         : manager(manager),
           section(section),
           id(id),
-          updateEvent(std::make_shared<kernel::type::KEvent>(manager.state, false)) {}
+          updateEvent(std::make_shared<kernel::type::KEvent>(manager.state, false)) {
+
+        // HW initializes the first 19 entries
+        for (std::size_t i = 0; i < 19; ++i) {
```
Comment (bylaws) [inline app/src/main/cpp/skyline/input/npad_device.cpp:16]: `size_t i{}` and maybe split out 19 into a static constexpr constant

**Review by bylaws — CHANGES_REQUESTED** [review]

File: app/src/main/cpp/skyline/input/npad_device.cpp (hunk 2 above)
Comment (german77) [reply app/src/main/cpp/skyline/input/npad_device.cpp:203]: Yes. All HID devices have the same behavior

**Pull request merged**